### PR TITLE
[DYN-5013] Trust location added in Prefrences Panel to take effect immediately

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

https://jira.autodesk.com/browse/DYN-5013
this update will show and hide file trust warning when as it is added to the trusted locations and preference panel is closed to save settings.

DynamoSandbox_SxptJ5GDsX

DynamoSandbox_kUns6Qfij0

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

- Trust location added in Prefrences Panel to take effect immediately

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -1621,6 +1621,7 @@ private void Open(object parameters)
                     // Skip these when opening dyf
                     FileTrustViewModel.DynFileDirectoryName = directoryName;
                     FileTrustViewModel.ShowWarningPopup = true;
+                    FileTrustViewModel.AllowOneTimeTrust = false; ;
```
Comment (reddyashish) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:1624]: Formatting error: "; ;"

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 2]
```diff
@@ -2863,6 +2864,33 @@ private void ExportToSTL(object parameter)
             }
         }
 
+        /// <summary>
+        /// Hide or show file trust warning popup for the current workspace when preference settings are updated.
+        /// </summary>
+        internal void ShowHideFileTrustWarningIfCurrentWorkspaceTrusted()
```
[the remark this answers is not in the document]
Comment (zeusongit) [reply src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs:2870]: `if ((FileTrustViewModel.ShowWarningPopup` here we check if the warning is being displayed

**Review by reddyashish — APPROVED** [review]

**Comment by reddyashish** [comment]
There is one test failure.

File: test/DynamoCoreWpfTests/PreferencesGeometryScalingTests.cs [hunk 3]
```diff
@@ -7,6 +7,7 @@
 using Dynamo.Wpf.Views;
 using DynamoCoreWpfTests.Utility;
 using NUnit.Framework;
+using Dynamo.Models;
```
Comment (QilongTang) [inline test/DynamoCoreWpfTests/PreferencesGeometryScalingTests.cs:10]: hmm I see no code changes in this file.. Is this line required?

**Review by QilongTang — APPROVED** [review]

**Pull request merged**